### PR TITLE
Add golangci-lint

### PR DESCRIPTION
**Pull request by jhrozek**
- make: Add a Makefile target that runs golangci-lint

Adds a new make target 'make verify' that is meant to run all the linters and code checks we'll have. For now, there's a single dependency `make verify-go-lint` that runs the golangci-lint linter.

- Fix linter errors reported by golangci-lint

Please see its commit message for all the fixes

- Add a github action to run the linter on every commit

what the title says

### CODE REVIEW

**Comment by jhrozek** [comment]
note: I started with enabling all checks and disabling only those that I found really spurious. But we can also adjust the checks as we go if we find that they are holding us back..

File: Makefile [hunk 1]
```diff
@@ -27,7 +28,14 @@ test:
 test-migrate: $(TOOLS_DIR)/migrate
 	MIGRATE=$(MIGRATE) migrations/test.sh
 
-.PHONY: $(TOOLS_DIR)/migrate
```
Comment (jhrozek) [inline Makefile:30]: btw this probably doesn't belong to this patch, but it's a one-liner..the rule shouldn't be PHONY from the start because the rule target points to a file, PHONY rules should IIRC be used for rules that don't correspond to a file

Comment (rhmdnd) [reply Makefile:30]: Yep - had to look this up. You're right.

https://www.gnu.org/software/make/manual/html_node/Phony-Targets.html

**Review by rhmdnd — APPROVED** [review]
